config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -99,3 +113,212 @@ const (
+// - "Disabled": The Alertmanager instance will not be deployed.
+// - "DefaultConfig": The Alertmanager instance will be deployed with default settings.
+// - "CustomConfig": The Alertmanager instance will be deployed with custom configuration.
+// +kubebuilder:validation:Enum=Disabled;DefaultConfig;CustomConfig
+type AlertManagerDeployMode string
+
+const (
+	// AlertManagerModeDisabled means the Alertmanager instance will not be deployed.
+	AlertManagerDeployModeDisabled AlertManagerDeployMode = "Disabled"
+	// AlertManagerModeDefaultConfig means the Alertmanager instance will be deployed with default settings.
+	AlertManagerDeployModeDefaultConfig AlertManagerDeployMode = "DefaultConfig"
+	// AlertManagerModeCustomConfig means the Alertmanager instance will be deployed with custom configuration.
+	AlertManagerDeployModeCustomConfig AlertManagerDeployMode = "CustomConfig"
+)
+
+// logLevel defines the verbosity of logs emitted by Alertmanager.
+// Valid values are Error, Warn, Info and Debug.
+// +kubebuilder:validation:Enum=Error;Warn;Info;Debug
+type LogLevel string
+
+const (
+	// Error only errors will be logged.
+	LogLevelError LogLevel = "Error"
+	// Warn, both warnings and errors will be logged.
+	LogLevelWarn LogLevel = "Warn"
+	// Info, general information, warnings, and errors will all be logged.
+	LogLevelInfo LogLevel = "Info"
+	// Debug, detailed debugging information will be logged.
+	LogLevelDebug LogLevel = "Debug"
+)
+
+// ContainerResource defines a single resource requirement for a container.
+// +kubebuilder:validation:XValidation:rule="has(self.request) || has(self.limit)",message="at least one of request or limit must be set"
+// +kubebuilder:validation:XValidation:rule="!(has(self.request) && has(self.limit)) || quantity(self.limit).compareTo(quantity(self.request)) >= 0",message="limit should not be less than request"
+type ContainerResource struct {
+	// name of the resource (e.g. "cpu", "memory", "hugepages-2Mi").
+	// This field is required.
+	// name must consist only of alphanumeric characters, `-`, `_` and `.` and must start and end with an alphanumeric character.
+	// +required
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.qualifiedName().validate(self).hasValue()",message="name must consist only of alphanumeric characters, `-`, `_` and `.` and must start and end with an alphanumeric character"
+	Name string `json:"name"`
+
+	// request is the minimum amount of the resource required (e.g. "2Mi", "1Gi").
+	// This field is optional.
+	// When limit is specified, request cannot be greater than limit.
+	// +optional
+	// +kubebuilder:validation:XIntOrString
+	// +kubebuilder:validation:MaxLength=20
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:XValidation:rule="isQuantity(self) && quantity(self).isGreaterThan(quantity('0'))",message="request must be a non-negative quantity"
+	Request resource.Quantity `json:"request,omitempty"`
```

Comment:
I think to avoid the CEL cost explosion, we will need to change this to just a string and drop `+kubebuilder:validation:XIntOrString`

Reply:
I don't see anymore the CEL cost error after last changes.

Reply:
Which change do you believe fixed the issue?

Reply:
`/ +kubebuilder:validation:XValidation:rule="!(has(self.request) && has(self.limit)) || quantity(self.limit).compareTo(quantity(self.request)) >= 0",message="limit must be greater than or equal to request"`